### PR TITLE
Scheduler docstrings, typing, polish

### PR DESCRIPTION
**Pull request by erikkemperman**
Sorry, this is a pretty huge diff. But I tried to make sure that these changes are all basically trivial.

### CODE REVIEW

File: tests/test_concurrency/test_threadpoolscheduler.py [hunk 1]
```diff
@@ -56,22 +67,22 @@ def test_schedule_action_absolute(self):
...
             evt.set()
 
-        nt.schedule_absolute(datetime.utcnow()+timedelta(milliseconds=100), action)
+        nt.schedule_absolute(default_now() + timedelta(milliseconds=100), action)
         evt.wait()
 
     def test_schedule_action_cancel(self):
         nt = thread_pool_scheduler
-        set = []
+        set = False
```
Comment (jcafhe) [inline tests/test_concurrency/test_threadpoolscheduler.py]: Maybe it would be more relevant to rename `set` to  `ran`?

**Comment by jcafhe** [comment]
Huge diff indeed! 

It may be not the place to discuss this here, but I was wondering if we should take care of one corner case  in `schedule_absolute` or `schedule_relative`.  Basically, we can say that if duetime equals 0.0s, `schedule` can be used as a 'shortcut'. 
Should it be `duetime <= 0.0` instead, i.e. do you think duetime can be negative in some cases?

**Comment by erikkemperman** [comment]
@jcafhe Thanks for thorough review, extremely helpful!

I followed almost all of your suggestions, except the symmetric time diffs in the unit tests: I found that, especially on Travis, we have to allow a bit more room on the "right-hand side", as their VMs are competing for physical hardware.

I also agree about the "shortcut" options but am planning to address that separately (which I mentioned earlier, consolidating some login in SchedulerBase).

**Comment by jcafhe** [comment]
>  I found that, especially on Travis, we have to allow a bit more room on the "right-hand side", as their VMs are competing for physical resources.

Got it :+1: 

Good job!

**Comment by erikkemperman** [comment]
Thanks for review!

**Pull request merged**